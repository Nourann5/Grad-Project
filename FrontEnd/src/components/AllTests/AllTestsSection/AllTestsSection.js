import React, { useEffect, useState } from 'react'
import { Col, Container, Row } from 'react-bootstrap'
import styles from './AllTestsSection.module.css'

import PageCategoryCard from 'components/Global/Elements/PageCategoryCard/PageCategoryCard'

import {ReactComponent as ExamIcon} from 'assets/icons/examCat.svg'
import AllTestFilter from '../AllTestFilter/AllTestFilter'

function AllTestsSection() {
    const cardData = [
        {
            cardText:'Sat Test Name1',
            cardIcon:<ExamIcon/>,
            cardLink:'/test',
            cardFilter:'r,w,mwo'
        },
        {
            cardText:'Sat Test Name2',
            cardIcon:<ExamIcon/>,
            cardLink:'/test',
            cardFilter:'r,w,mw'
        },
        {
            cardText:'Sat Test Name3',
            cardIcon:<ExamIcon/>,
            cardLink:'/test',
            cardFilter:'mw,mwo'
        },
        {
            cardText:'Sat Test Name4',
            cardIcon:<ExamIcon/>,
            cardLink:'/test',
            cardFilter:'r'
        },
        {
            cardText:'Sat Test Name5',
            cardIcon:<ExamIcon/>,
            cardLink:'/test',
            cardFilter:'w'
        },
        {
            cardText:'Sat Test Name6',
            cardIcon:<ExamIcon/>,
            cardLink:'/test',
            cardFilter:'r,w,mw,mwo'
        },
        {
            cardText:'Sat Test Name7',
            cardIcon:<ExamIcon/>,
            cardLink:'/test',
            cardFilter:'mwo'
        }
    ]

    const [filterTests ,setFilterTests] =useState('all')
    const [filterData , setFilterData] =useState(cardData)
    let changeFilterData = (type)=>{
        setFilterTests(type)
    }
    useEffect(()=>{
        if(filterTests ==='all'){
            setFilterData(cardData)
        }else{
            let filterData = cardData.filter(card=>{
                return card.cardFilter.includes(filterTests)
            })
            setFilterData(filterData)
        }
    },[filterTests])
    return (
        <section className={styles["show-tests"]}>
            <Container fluid='lg'>
                <AllTestFilter changeFilterData={changeFilterData} filterTests={filterTests} />
                <div className={styles["show-tests__cats"]}>
                    <Row>
                        {
                            filterData && filterData.map((card,index)=>(
                                <Col lg='4' md='6' className='mb-4' key={card.cardText +index} data-filter={card.cardFilter}>
                                    <PageCategoryCard cardInfo={card} />
                                </Col>
                            ))
                        }
                    </Row>
                </div>
            </Container>
        </section>
    )
}

export default AllTestsSection