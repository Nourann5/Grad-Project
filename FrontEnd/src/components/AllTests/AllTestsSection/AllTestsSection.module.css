.show-tests{
    min-height: calc(100vh - 174px);
}

.show-tests__cats{
    display:flex;
    align-items: center;
    justify-content: space-around;
}