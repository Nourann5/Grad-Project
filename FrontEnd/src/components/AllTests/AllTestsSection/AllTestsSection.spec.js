import {render , screen} from '@testing-library/react'
import { Provider } from 'react-redux'
import { BrowserRouter, Route } from 'react-router-dom'
import store from 'store/store'
import AllTestsSection from './AllTestsSection'

describe('Test the All Tests Section', () => { 
    it('Heading exists',()=>{
        const tree =  render(
            <BrowserRouter>
                <AllTestsSection />
            </BrowserRouter>
        )
        // const headerTitle = screen.getByRole('paragraph',{name:'Welcome Back'})
        // expect(headerTitle).toBeInTheDocument()
        expect(tree).toMatchSnapshot()
    })
})