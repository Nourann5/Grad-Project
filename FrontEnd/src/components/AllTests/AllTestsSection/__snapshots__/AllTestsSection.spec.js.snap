// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the All Tests Section Heading exists 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="show-tests"
      >
        <div
          class="container-lg"
        >
          <div
            class="show-categories__filter"
          >
            <span
              class="show-categories__filter-span"
            >
              Sort By
              <svg>
                sortBy.svg
              </svg>
            </span>
            <div
              class="show-categories__filter-items"
            >
              <button
                class="show-categories__filter-button btn 
                active"
              >
                All
              </button>
              <button
                class="show-categories__filter-button btn 
                "
              >
                Writing
              </button>
              <button
                class="show-categories__filter-button btn 
                "
              >
                Reading
              </button>
              <button
                class="show-categories__filter-button btn 
                "
              >
                Math Without Calc
              </button>
              <button
                class="show-categories__filter-button btn 
                "
              >
                Math With Calc
              </button>
            </div>
          </div>
          <div
            class="show-tests__cats"
          >
            <div
              class="row"
            >
              <div
                class="mb-4 col-lg-4 col-md-6"
                data-filter="r,w,mwo"
              >
                <div
                  class="show-categories__cat"
                >
                  <a
                    class="show-categories__cat-link"
                    href="/exams/undefined"
                  />
                  <div
                    class="show-categories__img-container"
                  >
                    <img
                      class="show-categories__img"
                    />
                    <h2
                      class="show-categories__cat-title"
                    />
                  </div>
                </div>
              </div>
              <div
                class="mb-4 col-lg-4 col-md-6"
                data-filter="r,w,mw"
              >
                <div
                  class="show-categories__cat"
                >
                  <a
                    class="show-categories__cat-link"
                    href="/exams/undefined"
                  />
                  <div
                    class="show-categories__img-container"
                  >
                    <img
                      class="show-categories__img"
                    />
                    <h2
                      class="show-categories__cat-title"
                    />
                  </div>
                </div>
              </div>
              <div
                class="mb-4 col-lg-4 col-md-6"
                data-filter="mw,mwo"
              >
                <div
                  class="show-categories__cat"
                >
                  <a
                    class="show-categories__cat-link"
                    href="/exams/undefined"
                  />
                  <div
                    class="show-categories__img-container"
                  >
                    <img
                      class="show-categories__img"
                    />
                    <h2
                      class="show-categories__cat-title"
                    />
                  </div>
                </div>
              </div>
              <div
                class="mb-4 col-lg-4 col-md-6"
                data-filter="r"
              >
                <div
                  class="show-categories__cat"
                >
                  <a
                    class="show-categories__cat-link"
                    href="/exams/undefined"
                  />
                  <div
                    class="show-categories__img-container"
                  >
                    <img
                      class="show-categories__img"
                    />
                    <h2
                      class="show-categories__cat-title"
                    />
                  </div>
                </div>
              </div>
              <div
                class="mb-4 col-lg-4 col-md-6"
                data-filter="w"
              >
                <div
                  class="show-categories__cat"
                >
                  <a
                    class="show-categories__cat-link"
                    href="/exams/undefined"
                  />
                  <div
                    class="show-categories__img-container"
                  >
                    <img
                      class="show-categories__img"
                    />
                    <h2
                      class="show-categories__cat-title"
                    />
                  </div>
                </div>
              </div>
              <div
                class="mb-4 col-lg-4 col-md-6"
                data-filter="r,w,mw,mwo"
              >
                <div
                  class="show-categories__cat"
                >
                  <a
                    class="show-categories__cat-link"
                    href="/exams/undefined"
                  />
                  <div
                    class="show-categories__img-container"
                  >
                    <img
                      class="show-categories__img"
                    />
                    <h2
                      class="show-categories__cat-title"
                    />
                  </div>
                </div>
              </div>
              <div
                class="mb-4 col-lg-4 col-md-6"
                data-filter="mwo"
              >
                <div
                  class="show-categories__cat"
                >
                  <a
                    class="show-categories__cat-link"
                    href="/exams/undefined"
                  />
                  <div
                    class="show-categories__img-container"
                  >
                    <img
                      class="show-categories__img"
                    />
                    <h2
                      class="show-categories__cat-title"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="show-tests"
    >
      <div
        class="container-lg"
      >
        <div
          class="show-categories__filter"
        >
          <span
            class="show-categories__filter-span"
          >
            Sort By
            <svg>
              sortBy.svg
            </svg>
          </span>
          <div
            class="show-categories__filter-items"
          >
            <button
              class="show-categories__filter-button btn 
                active"
            >
              All
            </button>
            <button
              class="show-categories__filter-button btn 
                "
            >
              Writing
            </button>
            <button
              class="show-categories__filter-button btn 
                "
            >
              Reading
            </button>
            <button
              class="show-categories__filter-button btn 
                "
            >
              Math Without Calc
            </button>
            <button
              class="show-categories__filter-button btn 
                "
            >
              Math With Calc
            </button>
          </div>
        </div>
        <div
          class="show-tests__cats"
        >
          <div
            class="row"
          >
            <div
              class="mb-4 col-lg-4 col-md-6"
              data-filter="r,w,mwo"
            >
              <div
                class="show-categories__cat"
              >
                <a
                  class="show-categories__cat-link"
                  href="/exams/undefined"
                />
                <div
                  class="show-categories__img-container"
                >
                  <img
                    class="show-categories__img"
                  />
                  <h2
                    class="show-categories__cat-title"
                  />
                </div>
              </div>
            </div>
            <div
              class="mb-4 col-lg-4 col-md-6"
              data-filter="r,w,mw"
            >
              <div
                class="show-categories__cat"
              >
                <a
                  class="show-categories__cat-link"
                  href="/exams/undefined"
                />
                <div
                  class="show-categories__img-container"
                >
                  <img
                    class="show-categories__img"
                  />
                  <h2
                    class="show-categories__cat-title"
                  />
                </div>
              </div>
            </div>
            <div
              class="mb-4 col-lg-4 col-md-6"
              data-filter="mw,mwo"
            >
              <div
                class="show-categories__cat"
              >
                <a
                  class="show-categories__cat-link"
                  href="/exams/undefined"
                />
                <div
                  class="show-categories__img-container"
                >
                  <img
                    class="show-categories__img"
                  />
                  <h2
                    class="show-categories__cat-title"
                  />
                </div>
              </div>
            </div>
            <div
              class="mb-4 col-lg-4 col-md-6"
              data-filter="r"
            >
              <div
                class="show-categories__cat"
              >
                <a
                  class="show-categories__cat-link"
                  href="/exams/undefined"
                />
                <div
                  class="show-categories__img-container"
                >
                  <img
                    class="show-categories__img"
                  />
                  <h2
                    class="show-categories__cat-title"
                  />
                </div>
              </div>
            </div>
            <div
              class="mb-4 col-lg-4 col-md-6"
              data-filter="w"
            >
              <div
                class="show-categories__cat"
              >
                <a
                  class="show-categories__cat-link"
                  href="/exams/undefined"
                />
                <div
                  class="show-categories__img-container"
                >
                  <img
                    class="show-categories__img"
                  />
                  <h2
                    class="show-categories__cat-title"
                  />
                </div>
              </div>
            </div>
            <div
              class="mb-4 col-lg-4 col-md-6"
              data-filter="r,w,mw,mwo"
            >
              <div
                class="show-categories__cat"
              >
                <a
                  class="show-categories__cat-link"
                  href="/exams/undefined"
                />
                <div
                  class="show-categories__img-container"
                >
                  <img
                    class="show-categories__img"
                  />
                  <h2
                    class="show-categories__cat-title"
                  />
                </div>
              </div>
            </div>
            <div
              class="mb-4 col-lg-4 col-md-6"
              data-filter="mwo"
            >
              <div
                class="show-categories__cat"
              >
                <a
                  class="show-categories__cat-link"
                  href="/exams/undefined"
                />
                <div
                  class="show-categories__img-container"
                >
                  <img
                    class="show-categories__img"
                  />
                  <h2
                    class="show-categories__cat-title"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
